### PR TITLE
added safeToString() to prevent credentials from being written to log

### LINKED ISSUE #469: In Client.bind() password gets logged at trace level
If I set the logger to trace level, the password gets logged.
I don't know if this is expected, but I think the password should be excluded from the logger at any level.

```
var ldap = require('ldapjs');
var Logger = require('bunyan');

log = new Logger({ name: 'ldap', component: 'client', stream: process.stderr, serializers: Logger.stdSerializers, level: 10 });
var client = ldap.createClient({ url: 'ldap://ldapserver', log: log });
client.bind('some_user', 'SeCrEt', (err) => {})
```

outputs:
```
{"name":"ldap","component":"client","hostname":"<thishost>","pid":15748,"clazz":"Client","ldap_id":"2__ldap://<ldapserver>","level":10,"msg":"sending request {\"messageID\":2,\"protocolOp\":\"BindRequest\",\"version\":3,\"name\":\"some_user\",\"authenticationType\":\"Simple\",\"credentials\":\"SeCrEt\",\"controls\":[]}","time":"2018-02-19T14:47:35.573Z","v":0}
```

ldapjs: v1.0.1

### PR DESCRIPTION
**Pull request by mgoldsborough**
Fixes #469.

### CODE REVIEW

**Comment by Barathwaja** [comment]
Is this still open ? any plans on merging?

**Comment by mgoldsborough** [comment]
@Barathwaja never received any feedback on this. I've since moved on to another means of authentication.

File: lib/client/client.js [hunk 1]
```diff
@@ -1469,8 +1469,9 @@ Client.prototype._sendSocket = function _sendSocket(message,
     timer = setTimeout(onRequestTimeout, self.timeout);
   }
 
-  if (log.trace())
-    log.trace('sending request %j', message.json);
+  if (log.trace()) {
+    log.trace('sending request %j', message.safeToString());
```
Comment (jsumners) [inline lib/client/client.js]: Where is the `safeToString` method defined? I don't see it when I search the existing code base.

File: lib/messages/bind_request.js [hunk 2]
```diff
@@ -71,13 +71,18 @@ BindRequest.prototype._toBer = function (ber) {
   return ber;
 };
 
-BindRequest.prototype._json = function (j) {
+BindRequest.prototype._json = function (j, safePrint) {
```
Comment (jsumners) [inline lib/messages/bind_request.js]: Can you show me some instances where this method is invoked? Are invokers already passing this second parameter and it just wasn't implemented? Or is this a completely new parameter that would need to be passed from somewhere?

**Comment by jsumners** [comment]
⚠️ This issue has been locked due to age. If you have encountered a recent
problem that seems to be covered by this issue, please open a new issue.

Please include a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)
when opening a new issue.

**Pull request closed without merging**